### PR TITLE
added a per-OH hard reset remove function

### PR DESCRIPTION
**Pull request by evka85**
the new fpga_remove_hard_reset() function removes the hard reset for
selected OHs

## Description

fpga_keep_hard_reset() function asserts hard reset and keeps it asserted
for selected OHs
the new fpga_remove_hard_reset() function removes the hard reset for
selected OHs
If sending a hard reset to a specific set of OHs is desired, one should
call fpga_keep_hard_reset() and then call fpga_remove_hard_reset() with
the same OHlist

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Sending a single hard reset to specific OHs is desired by the operations team

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bdorney** [comment]
can you integrate this into `sca.py` under [fpgaHeldInHardReset(args)](https://github.com/cms-gem-daq-project/reg_utils/blob/60e4e25a839bd521c5ccee4e99c8f83914f8a783/python/reg_interface/scripts/sca.py#L31-L34)

**Review by mexanick — CHANGES_REQUESTED** [review]
Could you please add the interface part as well?
It should be added here:
https://github.com/cms-gem-daq-project/reg_utils/blob/68391c2a1727c77e1e8c5c9b519c3cc0db9aa6e5/python/reg_interface/scripts/sca.py

**Comment by evka85** [comment]
Should we just change the functionality of the existing fpgaHeldInHardReset() function and the corresponding hh command, or should I introduce a new user function for this?
I guess there's not much need for asserting the hard reset and keeping it asserted, so we could just as well redefine the functionality of the hh command, but let me know your thoughts on this.

**Comment by evka85** [comment]
I suppose the h function could be named something like a "synchronous hard-reset done on all OHs connected to this AMC at the same time", and the hh function could be described as "manual hard-reset on the selected OHs defined by the ohMask, which is not guaranteed to arrive to all selected OHs at the same time"

**Comment by mexanick** [comment]
I'm fine with redefining the hh command. I don't recall any significant usage of it lately.

**Comment by evka85** [comment]
Ok, hmm I have a new commit, but can't see how to include it in this pull request.... Can you help?

**Comment by evka85** [comment]
Should I just delete this PR and create a new one?..

**Comment by mexanick** [comment]
no. Simply push your local commit. It will be added to this PR automatically

**Review by mexanick — APPROVED** [review]

File: python/reg_interface/common/sca_utils.py [hunk 1]
```diff
@@ -41,6 +41,13 @@ def fpga_keep_hard_reset(ohList):
     subheading('Asserting FPGA Hard Reset (and keeping it in reset)')
     sendScaCommand(ohList, 0x2, 0x10, 0x4, 0x0, False)
 
+def fpga_remove_hard_reset(ohList):
+    subheading('Disabling monitoring')
+    writeReg(getNode('GEM_AMC.SLOW_CONTROL.SCA.ADC_MONITORING.MONITORING_OFF'), 0xffffffff)
```
Comment (bdorney) [inline python/reg_interface/common/sca_utils.py]: This alters the monitoring, why?  Moreover after issuing the `sendSCACommand` at line 49 why doesn't it restore monitoring back to the original value?

Ideally it should:

1. read the setting of `GEM_AMC.SLOW_CONTROL.SCA.ADC_MONITORING.MONITORING_OFF` and store it in a temp value,
2. write `0xffffffff` to that register,
3. perform the `sendScaCommand` as shown, and then
4. restore `GEM_AMC.SLOW_CONTROL.SCA.ADC_MONITORING.MONITORING_OFF` to the original stored value.

**Review by bdorney — CHANGES_REQUESTED** [review]

File: python/reg_interface/common/sca_utils.py (hunk 1 above)
Comment (evka85) [reply python/reg_interface/common/sca_utils.py]: I agree, but this was always the case, so it's not related to this pull request.
We never used monitoring, so I just disabled it before running most of the SCA functions. But I agree that it should be read first, then disabled, and then restored to the original value at the end of a given function.

Comment (bdorney) [reply python/reg_interface/common/sca_utils.py]: Please implement steps 1 through 4 for this function `fpga_remove_hard_reset`.

Comment (evka85) [reply python/reg_interface/common/sca_utils.py]: Fine, I just pushed it. But it's completely untested, please test it.

**Review by bdorney — APPROVED** [review]

**Comment by evka85** [comment]
The continuous integration check failed, but it seems it's due to a reason unrelated to my changes?
Makefile:2: *** "Error: PETA_STAGE environment variable not set.  Change to the root of your PetaLinux install, and source the settings.sh file".  Stop.

**Comment by mexanick** [comment]
@evka85 yeah, this is due to issues with cross-compilation

**Pull request merged**